Mage.Sets/src/mage/cards/w/WitheringWisps.java

```diff
@@ -0,0 +1,149 @@
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.w;
+
+import java.util.UUID;
+import mage.abilities.ActivatedAbilityImpl;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.common.OnEventTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.condition.common.CreatureCountCondition;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.decorator.ConditionalTriggeredAbility;
+import mage.abilities.effects.common.DamageEverythingEffect;
+import mage.abilities.effects.common.SacrificeSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.TargetController;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.mageobject.SubtypePredicate;
+import mage.filter.predicate.mageobject.SupertypePredicate;
+import mage.filter.predicate.permanent.ControllerPredicate;
+import mage.util.CardUtil;
+
+/**
+ *
+ * @author L_J
+ */
+public class WitheringWisps extends CardImpl {
+
+    private static final String ruleText = "At the beginning of the end step, if no creatures are on the battlefield, sacrifice {this}.";
+
+    public WitheringWisps(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.ENCHANTMENT},"{1}{B}{B}");
+
+        // At the beginning of the end step, if no creatures are on the battlefield, sacrifice Withering Wisps.
+        TriggeredAbility triggered = new OnEventTriggeredAbility(GameEvent.EventType.END_TURN_STEP_PRE, "beginning of the end step", true, new SacrificeSourceEffect());
+        this.addAbility(new ConditionalTriggeredAbility(triggered, new CreatureCountCondition(0, TargetController.ANY), ruleText));
+
+        // {B}: Withering Wisps deals 1 damage to each creature and each player. Activate this ability no more times each turn than the number of snow Swamps you control.
+        this.addAbility(new WitheringWispsActivatedAbility());
+    }
+
+    public WitheringWisps(final WitheringWisps card) {
+        super(card);
+    }
+
+    @Override
+    public WitheringWisps copy() {
+        return new WitheringWisps(this);
+    }
+}
+
+class WitheringWispsActivatedAbility extends ActivatedAbilityImpl {
+
+    static class ActivationInfo {
+
+        public int turnNum;
+        public int activationCounter;
+
+        public ActivationInfo(int turnNum, int activationCounter) {
+            this.turnNum = turnNum;
+            this.activationCounter = activationCounter;
+        }
+    }
```

Comment:
This can probably be rolled into the main Activated Ability class. I'll look more into it in the morning

Reply:
Actually screw it, if this works then who cares.

Reply:
If I didn't include it, it would keep complaining about ActivationInfo not being public. Of course it would be possible to just make it public in the Activated Ability class, but I didn't want to mess with that one too much. I was mostly wondering whether there could be a cleaner way to turn a dynamic value into the maxActivationsPerTurn int.

Reply:
I'm making a few modifications right now, a bunch of stuff could just be switched from private to protected.